tests/functional/openstackdataplanedeployment_controller_test.go

```diff
@@ -30,15 +41,53 @@ var _ = Describe("Dataplane Deployment Test", func() {
 			Namespace: namespace,
 			Name:      "dataplane-ansible-ssh-private-key-secret",
 		}
+		neutronOvnMetadataSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "neutron-ovn-metadata-agent-neutron-config",
+		}
+		novaNeutronMetadataSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "nova-metadata-neutron-config",
+		}
+		novaCellComputeConfigSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "nova-cell1-compute-config",
+		}
+		ceilometerConfigSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "ceilometer-compute-config-data",
+		}
+		dataplaneIPSetName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "edpm-compute-node-1",
+		}
+		dataplaneNetConfigName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "dataplane-netconfig",
+		}
 		err := os.Setenv("OPERATOR_SERVICES", "../../config/services")
 		Expect(err).NotTo(HaveOccurred())
 	})
 
 	When("A dataplaneDeployment is created with matching NodeSet", func() {
 		BeforeEach(func() {
-			DeferCleanup(th.DeleteInstance, CreateDataplaneNodeSet(dataplaneNodeSetName, DefaultDataPlaneNoNodeSetSpec()))
-			DeferCleanup(th.DeleteInstance, CreateDataplaneDeployment(dataplaneDeploymentName, DefaultDataPlaneDeploymentSpec()))
 			CreateSSHSecret(dataplaneSSHSecretName)
+			th.CreateSecret(neutronOvnMetadataSecretName, map[string][]byte{
+				"authorized_keys": []byte("blih"),
```

Comment:
Actually, it will. We are using this value in our controller as on of the secrets.[0] If spec isn't declared preprovisioned the controller checks for this key. If it can't find it the reconciliation loop fails. 

[0] https://github.com/openstack-k8s-operators/dataplane-operator/blob/f3a157ac547c3290f0186b02eab4534857161d0d/controllers/openstackdataplanenodeset_controller.go#L88

Reply:
```
$ oc get OpenStackDataPlaneServices/ovn -o yaml
apiVersion: dataplane.openstack.org/v1beta1
kind: OpenStackDataPlaneService
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"dataplane.openstack.org/v1beta1","kind":"OpenStackDataPlaneService","metadata":{"annotations":{},"name":"ovn","namespace":"openstack"},"spec":{"configMaps":["ovncontroller-config"],"label":"dataplane-deployment-ovn","openStackAnsibleEERunnerImage":"quay.io/openstack-k8s-operators/openstack-ansibleee-runner:latest","playbook":"osp.edpm.ovn","secrets":["neutron-ovn-metadata-agent-neutron-config","nova-metadata-neutron-config"]}}
  creationTimestamp: "2023-10-02T09:41:38Z"
  generation: 2
  name: ovn
  namespace: openstack
  resourceVersion: "75183"
  uid: e48a079e-484d-4ce4-933b-6a0b899794d5
spec:
  configMaps:
  - ovncontroller-config
  label: dataplane-deployment-ovn
  playbook: osp.edpm.ovn
  secrets:
  - neutron-ovn-metadata-agent-neutron-config
  - nova-metadata-neutron-config
```

```
$ oc get Secret/neutron-ovn-metadata-agent-neutron-config -o yaml
apiVersion: v1
data:
  10-neutron-metadata.conf: IyBOT1RFOiBzb21lIG5lY2Vzc2FyeSBtZXRhZGF0YSBhZ2VudCBvcHRpb25zIGFyZSBzZXQgYnkgbm92YS4KIyBTcGVjaWZpY2FsbHksIG5vdmFfbWV0YWRhdGFfaG9zdCBhbmQgbWV0YWRhdGFfcHJveHlfc2hhcmVkX3NlY3JldC4KIwojIFRoZSBmaW5hbCBtZXRhZGF0YSBhZ2VudCBkZXBsb3ltZW50IHdpbGwgaGF2ZSB0byBtZXJnZSBzZXR0aW5ncyBmcm9tIG11bHRpcGxlCiMgc291cmNlcy4KCltvdm5dCm92bl9zYl9jb25uZWN0aW9uID0gdGNwOjE3Mi4xNy4wLjMwOjY2NDIK
kind: Secret
metadata:
  creationTimestamp: "2023-10-02T09:23:08Z"
  labels:
    neutron.openstack.org/name: neutron
    neutron.openstack.org/namespace: openstack
    neutron.openstack.org/uid: 6d46429a-208e-4b5f-92b1-9bd2e3a97696
  name: neutron-ovn-metadata-agent-neutron-config
  namespace: openstack
  ownerReferences:
  - apiVersion: neutron.openstack.org/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: NeutronAPI
    name: neutron
    uid: 6d46429a-208e-4b5f-92b1-9bd2e3a97696
  resourceVersion: "58082"
  uid: 2591ec65-65cb-439f-b32b-042fa7147abe
type: Opaque
```
No, It has a `10-neutron-metadata.conf` key.